packages/spindle-ui/src/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import MenuHorizotal from '../Icon/MenuHorizotal';
+
+import PaginationNavigation from './PaginationNavigation';
+import { usePageViewItems } from './hooks/usePageViewItems';
+
+interface Props extends React.HTMLAttributes<HTMLElement> {
+  current: number;
+  total: number;
+  isPageCount?: boolean;
+  isPagePrevNext?: boolean;
+  isPageFirstLast?: boolean;
+  handleClick: (pageNumber: number) => void;
+}
+
+const BLOCK_NAME = 'spui-Pagination';
+
+export const Pagination = (props: Props) => {
+  const {
+    current,
+    total,
+    isPageCount = false,
+    isPagePrevNext = true,
+    isPageFirstLast = false,
+    handleClick,
+    className,
+  } = props;
+
+  const pageItems = 5;
+  const { pageViewItems } = usePageViewItems({
+    current,
+    total,
+  });
+
+  // totalは表示数超えている前提で、前から2つ目のアイテムが2より大きいかどうか（最初が連続した数字じゃないことをチェック）
+  const viewPrevHorizotal = total > pageItems && 2 < pageViewItems[1];
+
+  // totalは表示数超えている前提で、後ろから2つ目のアイテムがtotal-1より小さいか（最後が連続した数字じゃないことをチェック）
+  const viewNextHorizotal =
+    total > pageItems && pageViewItems[pageItems - 2] < total - 1;
+
+  const handleOnClick = (pageNumber: number) => {
+    typeof handleClick === 'function' && handleClick(pageNumber);
+  };
+
+  return (
+    <nav
+      aria-label="ページネーション"
+      className={[BLOCK_NAME, className].filter(Boolean).join(' ').trim()}
+    >
+      <ul className={`${BLOCK_NAME}-list`}>
+        {isPageFirstLast && (
+          <PaginationNavigation
+            type="first"
+            current={current}
+            total={total}
+            handleButtonClick={(pageNumber) => handleOnClick(pageNumber)}
+          />
+        )}
+        {isPagePrevNext && (
+          <PaginationNavigation
+            type="prev"
+            current={current}
+            total={total}
+            handleButtonClick={(pageNumber) => handleOnClick(pageNumber)}
+          />
+        )}
+        {pageViewItems.map((item, index) => {
+          return (
+            <li
+              className={`${BLOCK_NAME}-item`}
+              key={`pagination-item-${index}`}
```

Comment:
```suggestion
              key={`pagination-item-${item}`}
```
itemを使っても良さそうな気がしました